Character movement

Codidodido opened the pull request:
Implement player movement with animations

Codidodido committed:
Implement player movement with animations

In 4 main directions

Codidodido committed:
Add collider to tiles and player

Work on appropriate colliders

Codidodido committed:
Implement Player running with animations

Codidodido committed:
Fix bug movement

Fix bug movement for moving in 4 sub directions because it added more speed to main speed

Codidodido committed:
Implement player running in sub directions

Codidodido committed:
Add energy feature to player

Decrease energy when running and increase energy when standing

Codidodido committed:
Fix silly animations bug

URAmiRBin commented:
fuck you

URAmiRBin commented on Assets/Animation/Player/RunBackwardPlayerAnimation.anim:
Move this into a folder

URAmiRBin commented on Assets/Scenes/SampleScene.unity:
Change the name

URAmiRBin commented on Assets/PlayerMovement.cs:
Move to scripts folder

URAmiRBin commented on Assets/PlayerMovement.cs:
```diff
@@ -0,0 +1,32 @@
+    [SerializeField] private float moveSpeed;
+    private Rigidbody2D _rigidbody2D;
+    private Vector2 _movement;
+    private Animator _animator;
```
`
[SerializeField] private float moveSpeed;

private Rigidbody2D _rigidbody2D;
private Vector2 _movement;
private Animator _animator;
`

Use proper spacing between methods and serialized and private fields

URAmiRBin commented on Assets/PlayerMovement.cs:
```diff
@@ -0,0 +1,32 @@
+    private Rigidbody2D _rigidbody2D;
+    private Vector2 _movement;
+    private Animator _animator;
+    private void Start()
```
Blank line on top of each method

Codidodido committed:
Organize directories

Codidodido committed:
Clean code

Remove extra spaces in PlayerMovement.cs code

URAmiRBin commented:
Please add pull request description for the reviewer to have a quick understanding of what is done in this branch

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+
+public class PlayerMovement : MonoBehaviour
+{
+    [Header("Player")]
```
this is redundant

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+    [SerializeField] private float moveSpeed;
+    [SerializeField] private float energy;
+    
+    private float _extraMoveSpeed;
```
the name is not clear, use `_deltaMoveSpeed`

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+    
+    private void Update()
+    {
+        _movement.x = Input.GetAxisRaw("Horizontal");
+        _movement.y = Input.GetAxisRaw("Vertical");
```
Wrap in function `HandleInput()`

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+        _movement.x = Input.GetAxisRaw("Horizontal");
+        _movement.y = Input.GetAxisRaw("Vertical");
+        
+        _animator.SetFloat("Horizontal",_movement.x);
+        _animator.SetFloat("Vertical",_movement.y);
+        _animator.SetFloat("Speed",Convert.ToInt32(_movement.magnitude));
```
Wrap in function `HandleMoveAnimation()`

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+        _movement.x = Input.GetAxisRaw("Horizontal");
+        _movement.y = Input.GetAxisRaw("Vertical");
+        
+        _animator.SetFloat("Horizontal",_movement.x);
```
```suggestion
        _animator.SetFloat("Horizontal", _movement.x);
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+
+    private void FixedUpdate()
+    {
+        _rigidbody2D.MovePosition(_rigidbody2D.position + _movement.normalized * ((moveSpeed+_extraMoveSpeed) * Time.fixedDeltaTime));
```
```suggestion
        _rigidbody2D.MovePosition(_rigidbody2D.position + _movement.normalized * (( moveSpeed + _extraMoveSpeed ) * Time.fixedDeltaTime));
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+                _animator.SetBool("IsRunning", true); 
+                _extraMoveSpeed = 0.5f;
+                DecreaseEnergy();
+            }else 
```
```suggestion
            } else 
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+    void IncreaseEnergy()
+    {
+        
+            if(energy <= 100f && _movement.sqrMagnitude ==0)
+            {
+                energy += 1f*Time.deltaTime;    
+            }
+            
```
Remove extra indentation

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+        
+            if(energy <= 100f && _movement.sqrMagnitude ==0)
+            {
+                energy += 1f*Time.deltaTime;    
```
```suggestion
                energy += 1f * Time.deltaTime;    
```

URAmiRBin commented on Assets/Scripts/PlayerMovement.cs:
```diff
@@ -0,0 +1,80 @@
+    void IncreaseEnergy()
+    {
+        
+            if(energy <= 100f && _movement.sqrMagnitude ==0)
```
This check should be done before calling `IncreaseEnergy()`

Codidodido committed:
Apply pull request #2 comments

Remove extra spaces, clear names, wrap some codes in functions, rename some variables
